src/Storages/MergeTree/IDataPartStorage.h

```diff
@@ -129,7 +129,11 @@ class IDataPartStorage
 
     /// Get a name like 'prefix_partdir_tryN' which does not exist in a root dir.
     /// TODO: remove it.
-    virtual std::string getRelativePathForPrefix(Poco::Logger * log, const String & prefix, bool detached) const = 0;
+    virtual std::optional<String> getRelativePathForPrefix(Poco::Logger * log, const String & prefix, bool detached, bool broken) const = 0;
+
+    virtual bool looksLikeBrokenDetachedPartHasTheSameContent(const String & detached_part_path,
```

Comment:
I would better remove this function from `IDataPartStorage.h` iface, only keep it in `DataPartStorageOnDisk`.
Also, comment is needed.

Reply:
Initially I tried to add this function to `IMergeTreeDataPart` instead of `IDataPartStorage`, but it's impossible because we have disk there.